TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/Pages/SupportTasks/ConnectOneLoginUser/Connect.cshtml.cs

```diff
@@ -29,17 +29,16 @@ public void OnGet()
 
     public async Task<IActionResult> OnPostAsync()
     {
-        var data = (ConnectOneLoginUserData)_supportTask!.Data;
-        _supportTask.Data = data with
+        var data = _supportTask!.UpdateData<ConnectOneLoginUserData>(data => data with
         {
             PersonId = PersonDetail!.PersonId
-        };
+        });
```

Comment:
Is this just a nicer way of doing the same thing?

Reply:
> Is this just a nicer way of doing the same thing?

Yes